### ISSUE TITLE
web_connectivity: resolver port incorrectly set

### ISSUE THREAD

**Issue by bassosimone** [measurement-kit/measurement-kit#744]
While testing #690, I noticed that the resolver port is incorrectly set in the results, e.g.:

```
...
                "failure": null, 
                "resolver_port": 8
            }
```

I believe this happens when only the resolver address is specified and there is no port.

Depends on #690.

To say it more abstractly, we need to decide the proper and right way to split host and port, write that implementation somewhere (assuming it's not already there), write regress tests, and then _consistently_ use it across MK.

**Comment by bassosimone** [comment]
I think we're probably going to fix this issue in `master`, rather than in `stable`, as part of a more comprehensive set of fixes triggered by #915. As such, I'm moving this issue from release-0.3.0 to release-0.4.0 and I am removing the "easy" tag (it is not easy because splitting when you find a `:` would break IPv6, plus one probably needs to account for IPv6 quoting).

**Comment by bassosimone** [comment]
We fixed this issue in #1079
